Week4/ReadingList/app/src/main/java/com/florencenjeri/readinglist/BooksFragment.kt

```diff
@@ -3,39 +3,41 @@ package com.florencenjeri.readinglist
 import android.os.Bundle
 import android.view.*
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
 import androidx.navigation.fragment.findNavController
-import androidx.recyclerview.widget.RecyclerView
 import com.florencenjeri.readinglist.model.Books
-import com.florencenjeri.readinglist.model.BooksData
+import com.florencenjeri.readinglist.model.UserPrefs
+import kotlinx.android.synthetic.main.books_fragment.view.*
 
-/**
- * A simple [Fragment] subclass as the default destination in the navigation.
- */
 class BooksFragment : Fragment(), BooksAdapter.BooksListClickListener {
-    val booksList = BooksData.booksRead
-    lateinit var booksRecyclerView: RecyclerView
+    private lateinit var booksList: List<Books>
```

Comment:
Do you need this booksList at all? You can just the adapter directly!

Reply:
I was using it as a hack to sort the list of data being retrieved from Room. How do I use the adapter directly to get the list?